packages/thirdweb/src/extensions/pack/createPack.test.ts

```diff
@@ -0,0 +1,187 @@
+import { mintTo as mintToERC721 } from "../erc721/write/mintTo.js";
+import { getPackContents } from "../erc1155/__generated__/IPack/read/getPackContents.js";
+import { getTokenCountOfBundle } from "../erc1155/__generated__/IPack/read/getTokenCountOfBundle.js";
+import { getUriOfBundle } from "../erc1155/__generated__/IPack/read/getUriOfBundle.js";
+import { openPack } from "../erc1155/__generated__/IPack/write/openPack.js";
+import { deployERC20Contract } from "../prebuilts/deploy-erc20.js";
+import { deployERC721Contract } from "../prebuilts/deploy-erc721.js";
+import { deployPackContract } from "../prebuilts/deploy-pack.js";
+import { createPack } from "./createPack.js";
+
+const account = TEST_ACCOUNT_A;
+const client = TEST_CLIENT;
+const chain = ANVIL_CHAIN;
+
+describe.runIf(process.env.TW_SECRET_KEY)("createPack", () => {
+  it("should work", async () => {
+    const packAddress = await deployPackContract({
+      account,
+      client,
+      chain,
+      params: {
+        name: "pack-contract",
+      },
+    });
+
+    const erc20Address = await deployERC20Contract({
+      client: TEST_CLIENT,
+      chain: ANVIL_CHAIN,
+      account: TEST_ACCOUNT_A,
+      type: "TokenERC20",
+      params: {
+        name: "Token",
+        contractURI: TEST_CONTRACT_URI,
+      },
+    });
+
+    const erc20Contract = getContract({ address: erc20Address, chain, client });
+
+    // Mint some ERC20 tokens
+    await sendAndConfirmTransaction({
+      transaction: mintToERC20({
+        contract: erc20Contract,
+        to: account.address,
+        amount: "100",
+      }),
+      account,
+    });
+
+    // Set allowance for Pack contract
+    await sendAndConfirmTransaction({
+      transaction: approve({
+        contract: erc20Contract,
+        amount: "1000000000000000",
+        spender: packAddress,
+      }),
+      account,
+    });
+
+    const erc721Address = await deployERC721Contract({
+      client: TEST_CLIENT,
+      chain: ANVIL_CHAIN,
+      account: TEST_ACCOUNT_A,
+      type: "TokenERC721",
+      params: {
+        name: "NFTCollection",
+        contractURI: TEST_CONTRACT_URI,
+      },
+    });
+
+    const erc721Contract = getContract({
+      address: erc721Address,
+      chain,
+      client,
+    });
+
+    // Mint some ERC721 tokens
+    await sendAndConfirmTransaction({
+      transaction: mintToERC721({
+        contract: erc721Contract,
+        to: account.address,
+        nft: { name: "token #0" },
+      }),
+      account,
+    });
+
+    // set erc721 approval
+    await sendAndConfirmTransaction({
+      transaction: setApprovalForAll({
+        contract: erc721Contract,
+        approved: true,
+        operator: packAddress,
+      }),
+      account,
+    });
+
+    const packContract = getContract({
+      address: packAddress,
+      chain,
+      client,
+    });
+
+    // Create pack
+    await sendAndConfirmTransaction({
```

Comment:
this could prob be separated in another test, and the things above in a beforeAll

Reply:
I intentionally avoid beforeAll and beforeEach because: https://thirdwebdev.slack.com/archives/C04LA5X7FNE/p1725035253820449

Reply:
manually setting the timeout value doesn't fix it...